docs: edit template-statements doc (#38742)
angular/angular: aio/content/guide/template-statements.md

</div>

In the following example, the template statement `deleteHero()` appears in quotes to the right of the `=`&nbsp;symbol as in `(event)="statement"`.

<code-example path="template-syntax/src/app/app.component.html" region="context-component-statement" header="src/app/app.component.html"></code-example>

> When the user clicks the "Delete hero" button, Angular calls the `deleteHero()` function in the component class.

Reviewer: ```suggestion
When the user clicks the **Delete hero** button, Angular calls the `deleteHero()` function in the component class.
```
Let's avoid using quote marks for emphasis. Bold for UI elements is fairly standard. Sound good?

You can use template statements with elements, components, or directives in response to events.

<div class="alert is-helpful">

Responding to events is an aspect of Angular's [unidirectional data flow](guide/glossary#unidirectional-data-flow).
You can change anything in your application during a single event loop.

</div>

## Syntax

Like [template expressions](guide/interpolation), template statements use a language that looks like JavaScript.

> However, the template statement parser differs from the template expression parser and specifically supports both basic assignment, `=`, and chaining expressions with semicolons, <code>;</code>.

Reviewer: ```suggestion
However, the parser for template statements differs from the parser for template expressions. In addition, the template statements parser specifically supports both basic assignment, `=`, and chaining expressions with semicolons, `;`.
```
This sentence seemed a bit difficult to read. I've offered a rewrite that breaks it into two. Also, any thoughts on why it had html (<code>) instead of markdown?

Author: There are some types of formatting in which the back ticks don't work, so at some point in the past it might have been within a section that had such formatting.

> Even so, the following JavaScript and template expression syntax is not allowed:

Reviewer: ```suggestion
The following JavaScript and template expression syntax is not allowed:
```
Phrases like "Even so" are sort of filler words. :)

## Statement context

Statements have a context&mdash;a particular part of the application to which the statement belongs.

Statements can refer only to what's in the statement context, which is typically the component instance.
For example, `deleteHero()` of `(click)="deleteHero()"` is a method of the component in the following snippet.

<code-example path="template-syntax/src/app/app.component.html" region="context-component-statement" header="src/app/app.component.html"></code-example>

The statement context may also refer to properties of the template's own context.
In the following example, the component's event handling method, `onSave()` takes the template's own `$event` object as an argument.
On the next two lines, the `deleteHero()` method takes a [template input variable](guide/built-in-directives#template-input-variable), `hero`, and `onSubmit()` takes a [template reference variable](guide/template-reference-variables), `#heroForm`.

<code-example path="template-syntax/src/app/app.component.html" region="context-var-statement" header="src/app/app.component.html"></code-example>

> Thus, the context of the `$event` object, `hero`, and `#heroForm` is the template.

Reviewer: ```suggestion
In this example, the context of the `$event` object, `hero`, and `#heroForm` is the template.
```
_Thus_ is a bit academic. :)

Template context names take precedence over component context names.

> In `deleteHero(hero)` above, the `hero` is the template input variable, not the component's `hero` property.

Reviewer: ```suggestion
In the preceding `deleteHero(hero)`, the `hero` is the template input variable, not the component's `hero` property.
```

## Statement best practices

* **Conciseness**

  Keep template statements minimal by using method calls or basic property assignments.

* **Work within the context**

  The context of a template statement can be the component class instance or the template.
  Because of this, template statements cannot refer to anything in the global namespace such as `window` or `document`.

>   Additionally, template statements can't call `console.log()` or `Math.max()`.

Reviewer 2: I think it'd be more accurate to use "For example" here, since the mentioned function calls are just examples of the global scope classes that can not be accessed within template statements:
```suggestion
  For example, template statements can't call `console.log()` or `Math.max()`.
```
